File: mall-frontend/src/api/merchant.ts
import http from './http'

/** 商家域接口：商家端（/api/merchant）+ 平台运营（/api/admin/merchants、/api/admin/withdrawals） */
export interface MerchantProfile {
  id: number
  merchantName: string
  contact?: string
  phone?: string
  shopName?: string
  shopLogo?: string
  shopDesc?: string
  shopAddress?: string
  shopStatus: number
  payCodeUrl?: string
  balance: number
  applyStatus: number
  status: number
}

export interface MerchantRow {
  id: number
  username: string
  merchantName: string
  contact?: string
  phone?: string
  applyStatus: number
  rejectReason?: string
  shopName?: string
  createTime: string
}

export interface Withdrawal {
  id: number
  withdrawalNo: string
  merchantId: number
  bankName: string
  accountNo: string
  holder: string
  amount: number
  status: number
  rejectReason?: string
  handleTime?: string
  createTime: string
}

interface Page<T> {
  records: T[]
  total: number
  current: number
  size: number
}

export const getShop = () => http.get<never, MerchantProfile>('/merchant/shop')
export const saveShop = (patch: Partial<MerchantProfile>) => http.put('/merchant/shop', patch)
export const applyWithdrawal = (data: { bankName: string; accountNo: string; holder: string; amount: number }) =>
  http.post('/merchant/withdrawal', data)
export const getMyWithdrawals = (page = 1, size = 10) =>
  http.get<never, Page<Withdrawal>>('/merchant/withdrawal', { params: { page, size } })

/** 图片上传（店铺 logo/收款码，MinIO 返回公开 URL，DB 存 URL） */
export const uploadMerchantImage = (file: File) => {
  const fd = new FormData()
  fd.append('file', file)
  return http.post<never, { url: string }>('/merchant/upload/image', fd, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

export const getMerchants = (applyStatus?: number, page = 1, size = 10) =>
  http.get<never, Page<MerchantRow>>('/admin/merchants', { params: { applyStatus, page, size } })
export const auditMerchant = (id: number, pass: boolean, reason?: string) =>
  http.put(`/admin/merchants/${id}/audit`, { pass, reason })

export const getWithdrawals = (status?: number, page = 1, size = 10) =>
  http.get<never, Page<Withdrawal>>('/admin/withdrawals', { params: { status, page, size } })
export const processWithdrawal = (id: number, pass: boolean, reason?: string) =>
  http.put(`/admin/withdrawals/${id}/process`, { pass, reason })

File: mall-backend/mall-merchant-service/src/main/java/com/mall/merchant/entity/Merchant.java
package com.mall.merchant.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.mall.common.BaseDO;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/** 商家（账号+入驻审核+店铺+收款码合并，物理保留）。写路径在本服务（认证字段 auth 侧只读校验）。 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("merchant")
public class Merchant extends BaseDO {

    private String username;

    private String password;

    /** 0 待审核 1 通过 2 驳回 */
    private Integer applyStatus;

    private String rejectReason;

    private LocalDateTime auditTime;

    private String merchantName;

    private String contact;

    private String phone;

    private String shopName;

    private String shopLogo;

    private String shopDesc;

    private String shopAddress;

    /** 0 营业 1 停业 */
    private Integer shopStatus;

    private String payCodeUrl;

    /** 0 正常 1 禁用 */
    private Integer status;

    /** 可提现余额：支付成功+/提现申请-/提现驳回+ */
    private BigDecimal balance;
}
